File: app/src/main/java/com/dicoding/InsightCart/MenuAdapter.java
package com.dicoding.InsightCart;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class MenuAdapter extends RecyclerView.Adapter<MenuAdapter.MenuViewHolder> {
    private Context context;
    private List<MenuItem> menuList;
    private OnItemClickListener listener;

    public interface OnItemClickListener {
        void onDeleteClick(int position);
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        this.listener = listener;
    }

    public static class MenuViewHolder extends RecyclerView.ViewHolder {
        public TextView menuTextView;
        public TextView priceTextView;
        public TextView actionTextView;
        public ImageButton deleteButton;

        public MenuViewHolder(@NonNull View itemView, final OnItemClickListener listener) {
            super(itemView);
            menuTextView = itemView.findViewById(R.id.menuTextView);
            priceTextView = itemView.findViewById(R.id.priceTextView);
            deleteButton = itemView.findViewById(R.id.deleteButton);

            deleteButton.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null) {
                        int position = getAdapterPosition();
                        if (position != RecyclerView.NO_POSITION) {
                            listener.onDeleteClick(position);
                        }
                    }
                }
            });
        }
    }

    public MenuAdapter(Context context, List<MenuItem> menuList) {
        this.context = context;
        this.menuList = menuList;
    }

    @NonNull
    @Override
    public MenuViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_recycler_view, parent, false);
        return new MenuViewHolder(view, listener);
    }

    @Override
    public void onBindViewHolder(@NonNull MenuViewHolder holder, int position) {
        MenuItem currentItem = menuList.get(position);
        holder.menuTextView.setText(currentItem.getMenu());
        holder.priceTextView.setText(currentItem.getPrice());
    }

    @Override
    public int getItemCount() {
        return menuList.size();
    }
}

File: app/src/main/java/com/dicoding/InsightCart/ui/Menu/MenuAdapter.kt
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.dicoding.InsightCart.R

class MenuAdapter(private val menuItems: List<MenuResponseItem>) :
    RecyclerView.Adapter<MenuAdapter.MenuViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MenuViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_menu, parent, false)
        return MenuViewHolder(view)
    }

    override fun onBindViewHolder(holder: MenuViewHolder, position: Int) {
        val menuItem = menuItems[position]
        holder.bind(menuItem)
    }

    override fun getItemCount(): Int {
        return menuItems.size
    }

    // ViewHolder untuk setiap item menu
    class MenuViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val textMenuName: TextView = itemView.findViewById(R.id.text_menu_name)
        private val textMenuPrice: TextView = itemView.findViewById(R.id.text_menu_price)

        fun bind(menuItem: MenuResponseItem) {
            textMenuName.text = menuItem.namaMenu
            textMenuPrice.text = "Rp.${menuItem.harga}"

            // Anda bisa bind data lainnya sesuai kebutuhan
        }
    }
}
